src/Altinn.Platform/Altinn.Platform.PDF/src/main/java/altinn/platform/pdf/services/PDFGenerator.java

```diff
@@ -214,19 +214,46 @@ private void renderFormLayout(List<FormLayoutElement> formLayout) throws IOExcep
     for (FormLayoutElement element : formLayout) {
       String componentType = element.getType();
       if (componentType.equalsIgnoreCase("group")) {
-        renderGroup(element, false);
+        if (LayoutUtils.includeComponentInPdf(element.getId(), layoutSettings)) {
+          if (element.getDataModelBindings().get("group") == null) {
+            renderGroup(element);
+          } else {
+            renderRepeatingGroup(element, false);
+          }
+        }
       } else {
         renderLayoutElement(element);
       }
     }
   }
 
-  private void renderGroup(FormLayoutElement element, boolean childGroup) throws IOException {
-    String componentId = element.getId();
-    if (!LayoutUtils.includeComponentInPdf(componentId, layoutSettings)) {
-      return;
+  private void renderGroup(FormLayoutElement element) throws IOException {
+    for (String childId : element.getChildren()) {
+      FormLayoutElement childElement = originalFormLayout.getData().getLayout().stream().filter(formLayoutElement -> formLayoutElement.getId().equals(childId)).findFirst().orElse(null);
+
+      if(childElement == null) {
+        continue;
+      }
+
+      if (childElement.getType().equalsIgnoreCase("group")) {
+        if(childElement.getDataModelBindings().get("group") == null) {
```

Comment:
I guess `getDataModelBindings()` could also return null here, as nonrepeating groups don't have a databinding objects. Maybe we should add a check here as well. This might indicate that our java model is wrong too. 

This would have been so much easier in c#, where you could simply write 
`childElement?.getDataModelBindings()?.get("group") == null` 🙄

Reply:
I tested this with nonrepeating group, it returns null. So I don't think it's necessary, it's Java-magic you know ;) :)